frontend/src/components/labwork/overview/LabworkOverviewProtocolPanel.tsx

```diff
@@ -1,16 +1,31 @@
 import React from 'react'
-import { LabworkSummaryProtocol } from '../../../models/labwork_summary'
+import { LabworkStepGroup, LabworkSummaryProtocol } from '../../../models/labwork_summary'
 import LabworkOverviewStepGroup from './LabworkOverviewStepGroup'
 
 export interface LabworkProtocolPanelProps {
 	readonly protocol: LabworkSummaryProtocol
+	readonly hideEmptySteps: boolean
 }
 
-const LabworkOverviewProtocolPanel = ({ protocol }: LabworkProtocolPanelProps) => {
+function doesGroupHaveSamples(group: LabworkStepGroup) {
+	// count the number of samples across the group's steps
+	const numSamples = group.steps.reduce((acc, step) => {
```

Comment:
Not the most efficient way to do that. You can bail once you find a single sample or deal with them as booleans.